bindings/pyroot/pythonizations/python/ROOT/_pythonization/_rvec.py

```diff
@@ -82,6 +82,75 @@
 }
 
 
+def _get_cpp_type_from_numpy_type(dtype):
+    cpptypes = {"i4": "int", "u4": "unsigned int", "i8": "Long64_t", "u8": "ULong64_t", "f4": "float", "f8": "double"}
+
+    if not dtype in cpptypes:
+        raise RuntimeError("Object not convertible: Python object has unknown data-type '" + dtype + "'.")
+
+    return cpptypes[dtype]
+
+
+def _AsRVec(arr):
+    """
+    Adopt memory of a Python object with array interface using an RVec.
+
+    \param[in] self self object
+    \param[in] obj PyObject with array interface
+
+    This function returns an RVec which adopts the memory of the given
+    PyObject. The RVec takes the data pointer and the size from the array
+    interface dictionary.
+    """
+    import ROOT
+    import cppyy
+    import math
+    import platform
+
+    # Get array interface of object
+    interface = arr.__array_interface__
+
+    # Get the data-pointer
+    data = interface["data"][0]
+
+    # Get the size of the contiguous memory
+    shape = interface["shape"]
+    size = math.prod(shape) if len(shape) > 0 else 0
+
+    # Get the typestring and properties thereof
+    typestr = interface["typestr"]
+    if len(typestr) != 3:
+        raise RuntimeError(
+            "Object not convertible: __array_interface__['typestr'] returned '"
+            + typestr
+            + "' with invalid length unequal 3."
+        )
+
+    if typestr[:1] != GetEndianess():
+        raise RuntimeError(
+            "Object not convertible: Endianess of __array_interface__['typestr'] does not match endianess of ROOT."
+        )
+
+    dtype = typestr[1:]
+    cppdtype = _get_cpp_type_from_numpy_type(dtype)
+
+    # Construct an RVec of the correct data-type
+    klassname = "ROOT::VecOps::RVec<" + cppdtype + ">"
+
+    prefix = "0x" if platform.system() == "Windows" else ""
+
+    address = ROOT.gInterpreter.Calc(f"new {klassname}(reinterpret_cast<{cppdtype} *>({prefix}{data}), {size})")
+
+    # Bind the object to a Python-side proxy, taking ownership of the underlying C++ object
+    out = cppyy.bind_object(address, klassname)
+    ROOT.SetOwnership(out, True)
```

Comment:
Same comment as with `_rtensor`, it would be very nice if we can avoid the extra call to the interpreter